ocs_ci/ocs/replica_one.py

```diff
@@ -0,0 +1,199 @@
+
+
+def get_replica_1_osds() -> tuple[list[str], list[str]]:
+    """
+    Gets the names and IDs of OSD associated with replica1
+
+    Returns:
+        tuple[list[str], list[str]]: list with OSDs names, list with OSD IDs
+
+    """
+    replica1_osds_id = list()
+    replica1_osds = list()
+    all_osds = get_pods_having_label(label=OSD_APP_LABEL)
+    for domain in config.ENV_DATA["worker_availability_zones"]:
+        for osd in all_osds:
+            if osd["metadata"]["labels"]["ceph.rook.io/DeviceSet"] == domain:
+                replica1_osds.append(osd["metadata"]["name"])
+                replica1_osds_id.append(osd["metadata"]["labels"]["ceph-osd-id"])
+    log.info(replica1_osds)
+    return (replica1_osds, replica1_osds_id)
+
+
+def get_replica1_osd_deployment() -> list[str]:
+    """
+    Gets the names of OSD deployments associated with replica1
+
+    Returns:
+        list[str]: deployment names
+
+    """
+    dep_obj = OCP(kind=DEPLOYMENT)
+    deployments = dep_obj.get()["items"]
+    replica1_osd_deployment = list()
+    osd_deployment = list()
+    for deployment in deployments:
+        if (
+            "metadata" in deployment
+            and "labels" in deployment["metadata"]
+            and "app.kubernetes.io/name" in deployment["metadata"]["labels"]
+        ):
+            if deployment["metadata"]["labels"]["app.kubernetes.io/name"] == "ceph-osd":
+                osd_deployment.append(deployment)
+
+    for deployment in osd_deployment:
+        if (
+            deployment["metadata"]["labels"]["ceph.rook.io/DeviceSet"]
+            in config.ENV_DATA["worker_availability_zones"]
+        ):
+            log.info(deployment["metadata"]["name"])
+            replica1_osd_deployment.append(deployment["metadata"]["name"])
+
+    return replica1_osd_deployment
+
+
+def scaledown_deployment(deployments_name=list[str]) -> None:
+    """
+    Scale down deployments to 0
+
+    Args:
+        deployments_name (list[str]): list of deployment names.
+
+    """
+    deployment_obj = OCP(kind=DEPLOYMENT)
+    for deployment in deployments_name:
+        deployment_obj.exec_oc_cmd(f"scale deployment {deployment} --replicas=0")
+        log.info(f"scaling to 0: {deployment}")
+
+
+def count_osd_pods() -> int:
+    """Get OSDs count in a cluster
+
+    Returns:
+        int: number of OSDs in cluster
+
+    """
+    return len(get_pods_having_label(label=OSD_APP_LABEL))
+
+
+def delete_replica_1_sc() -> None:
+    """
+    Deletes storage class associated with replica1
+
+    """
+    sc_obj = OCP(kind=STORAGECLASS, resource_name=REPLICA1_STORAGECLASS)
+    try:
+        sc_obj.delete(resource_name=REPLICA1_STORAGECLASS)
+    except CommandFailed as e:
+        if "Error is Error from server (NotFound)" in str(e):
+            log.info(
+                f"{REPLICA1_STORAGECLASS} not found, assuming it was already deleted"
+            )
+        else:
+            log.error("Failed to delete storage class")
```

Comment:
under this condition (the `else`) the exception won't get thrown and the function will just return while the the SC failed to get deleted